### ISSUE TITLE
On standardise functions `continuous=False` doesn't work

### ISSUE THREAD

**Issue by gareth-j** [openghg/openghg#887]
At the moment passing `False` to `continuous` doesn't stop it failing, look at why.

**Comment by SutarPrasad** [comment]
Scenario: 

 If the list of files is supplied something like this:
          ['MHD-10magl_UKV_EUROPE_201709.nc',
           'MHD-10magl_UKV_EUROPE_201601.nc',
           'MHD-10magl_UKV_EUROPE_202005.nc',]
          
  and later 2 more new entries are added to the same list
  it should raise, for the first 3 files in the list below mentioned error:
          "This file has been uploaded previously with the filename : MHD-10magl_EUROPE_200401.nc - skipping."
          
  and further, it should standardise the newly added 2 files.



Input:
standardise_footprint(site="MHD",
                      height="10m",
                      domain="EUROPE",
                      model="NAME",
                      filepath=unordered_files_list.to_list(),
                      store="zarr2",
                      continuous="False",
                     )
                     
 Expected: To standardise the list of unordered files successfully without raising the error mentioned in the output.

But currently, it raises the below error and returns empty "{}"

Output:
But it raises ValueError: Continuous data with no gaps is expected but no time-period can be inferred. Run with continuous=False to remove this constraint.
"{}"

**Comment by rt17603** [comment]
Quick note: running with the following
```python
standardise_footprint(site="MHD",
height="10m",
domain="EUROPE",
model="NAME",
filepath=unordered_files_list.to_list(),
store="zarr2",
continuous="False",
)
```
will still be returning the same `ValueError`. This is because `continuous` is being set to the value of `"False"` which is a string rather than the boolean `False`.

**Comment by rt17603** [comment]
One bug I found on this was that an input period could not be specified if this could not be inferred from the data. This is fixed by PR #899 .

**Comment by rt17603** [comment]
The main issue here is with the new way that the data is being collated to be parsed.

['MHD-10magl_UKV_EUROPE_201709.nc',
'MHD-10magl_UKV_EUROPE_201601.nc',
'MHD-10magl_UKV_EUROPE_202005.nc',]

Reading the following files all at once means we don't have a continuous timeseries anymore - we have 2017-09 in 1H frequency and then 201601 in 1H frequency... With the current logic this means it is correct that it can't infer the frequency of the data in this case (and raising the `ValueError` is correct).

The overall question is if we want to refactor this to change the underlying assumptions so the frequency can be inferred in these types of cases. For situations where we had an ordered, continuous list this should still work as it did previously (though we could put in a test for that - low priority).

**Comment by brendan-m-murphy** [comment]
> One bug I found on this was that an input period could not be specified if this could not be inferred from the data. This is fixed by PR #899 .

Also, if the inferred period is different from the specified period, it raises an error. This might be what we want, but it makes it hard to add a lot of data at once. 

For instance, some of the MHD CO2 footprints have gaps longer than 1 hour between measurements (only one file I've seen so far). The gap is quite large (49 hours), so if we force the period to be 1 hour, we'll end up aligning the obs to this gap.

I guess the real problem is this gap...

**Comment by rt17603** [comment]
> > One bug I found on this was that an input period could not be specified if this could not be inferred from the data. This is fixed by PR #899 .
> 
> Also, if the inferred period is different from the specified period, it raises an error. This might be what we want, but it makes it hard to add a lot of data at once.
> 
> For instance, some of the MHD CO2 footprints have gaps longer than 1 hour between measurements (only one file I've seen so far). The gap is quite large (49 hours), so if we force the period to be 1 hour, we'll end up aligning the obs to this gap.
> 
> I guess the real problem is this gap...

Proposed approach from discussion:
 - could remove `continuous` flag as an input
 - focus on `period` input
    - if not specified, i.e. `period=None`, try to infer the time period from the frequency of the data / 1 time point use the filename convention check.
        - raises error?
        - or could allow to pass through with `period="multiple"` or `period="varies"`?
    - if specified, e.g. `period="1H"`, check data is not *more* frequent than this (raise error?) then use this value

Can make it so that period *must* be either specified or inferred in this way.

**Comment by rt17603** [comment]
PR #899 included as a short term fix for some of the immediate issues.

Issue #911 opened to address other issues discussed above.
